### PR TITLE
Change how rest shape is given in RestShapeSpringsForceField [BREAKING]

### PR DESCRIPTION
**Pull request by raffaellatrivisonne**
In RestShapeSpringsForceField the rest_shape is currently provided using  a raw pointer which is really Bad. 
In this PR is change that to use SingleLink instead. 

The PR imply a change by which the syntax of the external_rest_shape is provided. Adding an "@" as in the test examples/Components/forcefield/RestShapeSpringsForceField2.scn make it work.

Editting: Damien. 
______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by guparan** [comment]
Hey @raffaellatrivisonne, congrats for your first PR ! 
Let's [ci-build] it...

File: SofaKernel/modules/SofaDeformable/RestShapeSpringsForceField.inl [hunk 1]
```diff
@@ -50,12 +50,11 @@ RestShapeSpringsForceField<DataTypes>::RestShapeSpringsForceField()
     , stiffness(initData(&stiffness, "stiffness", "stiffness values between the actual position and the rest shape position"))
     , angularStiffness(initData(&angularStiffness, "angularStiffness", "angularStiffness assigned when controlling the rotation of the points"))
     , pivotPoints(initData(&pivotPoints, "pivot_points", "global pivot points used when translations instead of the rigid mass centers"))
-    , external_rest_shape(initData(&external_rest_shape, "external_rest_shape", "rest_shape can be defined by the position of an external Mechanical State"))
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaDeformable/RestShapeSpringsForceField.inl:53]: Shouldn't the extern_rest_shape data field be removed ?

**Comment by damienmarchal** [comment]
Hi @raffaellatrivisonne 

Many thanks for your PR. You are totally right, SingleLink are far better.

**Comment by fjourdes** [comment]
This is indeed waaaaaay better, but be aware that it will actually break the scenes that were specifying a path in `RestShapeSpringForceField` using the prior implementation, since it was: 
- relying on `Data<std::string>` to store the path to the `MechanicalObject` instance whose positions are used as attachments
- on top of that (as far as I can remember ) the path passed as an argument was requiring some additional `..` directive to move in the node hierarchy, compared to the common practice, which ended up making things extremely confusing when using this particular component. 

So... this PR breaks things, but that is for the best.

**Comment by damienmarchal** [comment]
@fjourdes I fully agree with your comment...so I just made PR #322 that implements a global mecanism to handle these kind of situation. 
It is far from perfect but I rather like it compared to the existing way of managing changes in Sofa.

**Comment by fredroy** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by untereiner** [comment]
@raffaellatrivisonne could you delete your branch (push the below "delete branch" button) to clean the mimesis-inria repository. Thanks!

**Pull request merged**